auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue

```diff
@@ -371,6 +373,75 @@ export default class AffiliatedEntityTable extends Mixins(DateMixin) {
     return this.selectedColumns.includes(col)
   }
 
+  /** Custom sorting method to handle consolidated Nr and Affiliations data. */
+  private customSort (items, index, isDesc): any {
+    items.sort((a, b) => {
+      switch (index[0]) {
+        case 'lastModified':
+          let dateA, dateB
+          if (a.lastModified) {
+            dateA = a.lastModified
+          } else {
+            dateA = a.modified
+          }
+          if (b.lastModified) {
+            dateB = b.lastModified
+          } else {
+            dateB = b.modified
+          }
+
+          if (!isDesc[0]) {
+            return +new Date(dateB) - +new Date(dateA)
+          } else {
+            return +new Date(dateA) - +new Date(dateB)
+          }
+        case 'name':
+          let nameA, nameB
+          if (a.nameRequest) {
+            nameA = a.nameRequest?.names[0].name
+          } else {
+            nameA = this.isNumberedIncorporationApplication(a) ? 'Numbered Benefit Company' : a.name
+          }
+          if (b.nameRequest) {
+            nameB = b.nameRequest?.names[0].name
+          } else {
+            nameB = this.isNumberedIncorporationApplication(b) ? 'Numbered Benefit Company' : b.name
+          }
+
+          if (!isDesc[0]) {
+            return nameA.toLowerCase().localeCompare(nameB.toLowerCase())
+          } else {
+            return nameB.toLowerCase().localeCompare(nameA.toLowerCase())
+          }
+        case 'number':
+          if (!isDesc[0]) {
+            return this.number(a).toLowerCase().localeCompare(this.number(b).toLowerCase())
+          } else {
+            return this.number(b).toLowerCase().localeCompare(this.number(a).toLowerCase())
+          }
+        case 'type':
+          if (!isDesc[0]) {
+            return this.type(a).toLowerCase().localeCompare(this.type(b).toLowerCase())
+          } else {
+            return this.type(b).toLowerCase().localeCompare(this.type(a).toLowerCase())
+          }
+        case 'status':
+          if (!isDesc[0]) {
+            return this.status(a).toLowerCase().localeCompare(this.status(b).toLowerCase())
+          } else {
+            return this.status(b).toLowerCase().localeCompare(this.status(a).toLowerCase())
+          }
+        case 'modifiedBy':
+          if (!isDesc[0]) {
+            return this.modifiedBy(a).toLowerCase().localeCompare(this.modifiedBy(b).toLowerCase())
+          } else {
+            return this.modifiedBy(b).toLowerCase().localeCompare(this.modifiedBy(a).toLowerCase())
```

Comment:
may be we could do this[param].towLowerCase().localComapre(this[param]).toLowercase to avoid these duplication..just a suggestion.. all these 5 conditions can be one statement then

Reply:
may be similar approach can be used for above statments

Reply:
Okie doke! You bet